harness/determined/common/api/_session.py

```diff
@@ -174,9 +224,11 @@ def put(
 
     def with_retry(self: T, max_retries: GeneralizedRetry) -> T:
         """Generate a new session with a different retry policy."""
-        new_session = copy.copy(self)
-        new_session._max_retries = max_retries
-        return new_session
+        self._max_retries = max_retries
+        if self._http_session:
+            self._http_session.close()
+            self._http_session = self._make_http_session()
+        return self
```

Comment:
this changes the semantics a bit. I think the old method and `with_` naming and its docstring indicated you make a new session like the old one but with this one setting different. you can continue using the old session object with the old setting.

now, the underlying object is updated in place with the new setting. it does not seem like this method is ever used in a way where the old object is still preserved. so the change in semantics is fine. but maybe 
1. update the docstring to reflect the different behavior
2. consider renaming it to `set_retry` or something to reflect it updates the thing itself